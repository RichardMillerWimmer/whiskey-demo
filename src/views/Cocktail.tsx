import React from 'react';
import CocktailBox from '../components/CocktailBox';

const Cocktail: React.FC = () => {
    

    return (
        <section>
           <h2 data-testid='CocktailId'>Cocktail View</h2>
           <CocktailBox />
           <CocktailBox />
           <CocktailBox />
        </section>
    )
}

export default Cocktail;