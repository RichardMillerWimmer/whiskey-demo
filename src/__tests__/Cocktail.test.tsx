import React from 'react';
import Cocktail from '../views/Cocktail';
import {shallow, mount} from 'enzyme';
import { render, fireEvent } from '@testing-library/react';
import { act } from 'react-dom/test-utils'

let container: any = null;

beforeEach(() => {
  container = document.createElement('section');
  document.body.appendChild(container);
});
afterEach(() => {
  container.remove();
  container = null;
});

it("Cocktail renders correctly", () => {
  const cocktailWrapper = shallow(
    <Cocktail />
  );
  expect(cocktailWrapper).toMatchSnapshot();
});

it("renders with heading", () => {
    act(() => {
      render(<Cocktail />, container);
    });
    expect(container.textContent).toBe("Landing View")});