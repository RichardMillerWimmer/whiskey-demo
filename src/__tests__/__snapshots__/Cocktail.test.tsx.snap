// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cocktail renders correctly 1`] = `ShallowWrapper {}`;
